pkg/reconciler/taskrun/resources/taskspec_test.go

```diff
@@ -233,12 +235,59 @@ func TestGetTaskData_ResolvedNilTask(t *testing.T) {
 			},
 		},
 	}
-	getTask := func(ctx context.Context, n string) (*v1beta1.Task, *v1beta1.RefSource, error) {
-		return nil, nil, nil
+	getTask := func(ctx context.Context, n string) (*v1beta1.Task, *v1beta1.RefSource, *trustedresources.VerificationResult, error) {
+		return nil, nil, nil, nil
 	}
 	ctx := context.Background()
 	_, _, err := resources.GetTaskData(ctx, tr, getTask)
 	if err == nil {
 		t.Fatalf("Expected error when unable to find referenced Task but got none")
 	}
 }
+
+func TestGetTaskData_VerificationResult(t *testing.T) {
+	tr := &v1beta1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "mytaskrun",
+		},
+		Spec: v1beta1.TaskRunSpec{
+			TaskRef: &v1beta1.TaskRef{
+				ResolverRef: v1beta1.ResolverRef{
+					Resolver: "foo",
+					Params: v1beta1.Params{{
+						Name: "bar",
+						Value: v1beta1.ParamValue{
+							Type:      v1beta1.ParamTypeString,
+							StringVal: "baz",
+						},
+					}},
+				},
+			},
+		},
+	}
+	sourceMeta := metav1.ObjectMeta{
+		Name: "task",
+	}
+	sourceSpec := v1beta1.TaskSpec{
+		Steps: []v1beta1.Step{{
+			Name:   "step1",
+			Image:  "ubuntu",
+			Script: `echo "hello world!"`,
+		}},
+	}
+
+	verificationResult := &trustedresources.VerificationResult{
+		VerificationResultType: trustedresources.VerificationError,
+		Err:                    trustedresources.ErrResourceVerificationFailed,
+	}
+	getTask := func(ctx context.Context, n string) (*v1beta1.Task, *v1beta1.RefSource, *trustedresources.VerificationResult, error) {
+		return &v1beta1.Task{
+			ObjectMeta: *sourceMeta.DeepCopy(),
+			Spec:       *sourceSpec.DeepCopy(),
+		}, nil, verificationResult, nil
+	}
+	r, _, _ := resources.GetTaskData(context.Background(), tr, getTask)
```

Comment:
It looks like this is ignoring a return value error. can you use t.fatalf instead?